doc/specs/stdlib_logger.md

```diff
@@ -0,0 +1,965 @@
+`read_only_error`      | an output unit did not have an `access` specifier of `'WRITE'` or `'READWRITE'`
+`unformatted_in_error` | the unit did not have a `form` of `'FORMATTED'`
+`unopened_in_error`    | the unit was not opened
+`write_fault`          | one of the writes to `log_units` failed
+
+## The derived type: logger_type
+
+### Status
+
+Experimental
+
+### Description
+
+Serves to define 'logger' variables to be used in reporting
+significant events encountered during the execution of a program.
+
+### Syntax
+
+`type([[stdlib_logger(module):logger_type(type)]]) :: variable`
+
+### Private attributes
+
+| Attribute        | Type          | Description                                     | Initial value
+|------------------|---------------|-------------------------------------------------|--------------
+| `add_blank_line` | Logical       | Flag to precede output with a blank line        | `.false.`
+| `indent_lines`   | Logical       | Flag to indent subsequent lines by four columns | `.true.`
+| `log_units`      | Integer array | List of I/O units used for output               | empty
+| `max_width`      | Integer       | Maximum column width of output                  | 0
+| `time_stamp`     | Logical       | Flag to precede output by a time stamp          | `.true.`
+| `units`          | Integer       | Count of the number of active output units      | 0
+
+## The `stdlib_logger` variable
+
+The module defines one public variable, `global_logger`, of type
+`logger_type`. As might be guessed from its name, `global_logger` is
+intended to serve as the default logger for use throughout an
+application.
+
+
+### Public `logger_type` methods
+
+The module defines twelve public procedures: one function and eleven
+subroutines.  The
+procedures are:
+
+Procedure            | Class      | Description
+---------------------|------------|------------
+`add_log_file`       | Subroutine | Opens a file using `newunit`, and adds the resulting unit to the `log_units` list
+`add_log_unit`       | Subroutine | Adds an existing unit to the `log_units` list
+`configure`          | Subroutine | Configures the details of the logging process
+`configuration`      | Subroutine | Reports the details of the logging configuration
+`log_error`          | Subroutine | Sends a message prepended by `'ERROR: '` optionally followed by a `stat` or `errmsg`
+`log_information`    | Subroutine | Sends a message prepended by `'INFO: '`
+`log_io_error`       | Subroutine | Sends a message prepended by `'I/O ERROR: '` optionally followed by an `iostat` or `iomsg`
+`log_message`        | Subroutine | Sends a message
+`log_text_error`     | Subroutine | Sends a message describing an error found in a line of text
+`log_units_assigned` | Function   | Returns the number of active I/O units in `log_units`
+`log_warning`        | Subroutine | Sends a message prepended by `'WARN: '`
+`remove_log_unit`    | Subroutine | Removes the `unit` number from the `log_units` array
```

Comment:
```suggestion
[`add_log_file`](./stdlib_logger.html#add_log_file-open-a-file-and-add-its-unit-to-self-log_units)       | Subroutine | Opens a file using `newunit`, and adds the resulting unit to the `log_units` list
[`add_log_unit`](./stdlib_logger.html#add_log_unit-add-a-unit-to-the-array-self-log_units)       | Subroutine | Adds an existing unit to the `log_units` list
[`configuration`](./stdlib_logger.html#configuration-report-a-loggers-configuration)      | Subroutine | Reports the details of the logging configuration
[`configure`](./stdlib_logger.html#configure-configure-the-logging-process)          | Subroutine | Configures the details of the logging process
[`log_error`](./stdlib_logger.html#log_error-writes-the-string-message-to-self-log_units)          | Subroutine | Sends a message prepended by `'ERROR: '` optionally followed by a `stat` or `errmsg`
[`log_information`](./stdlib_logger.html#log_information-writes-the-string-message-to-self-log_units)    | Subroutine | Sends a message prepended by `'INFO: '`
[`log_io_error`](./stdlib_logger.html#log_io_error-write-the-string-message-to-self-log_units)       | Subroutine | Sends a message prepended by `'I/O ERROR: '` optionally followed by an `iostat` or `iomsg`
[`log_message`](./stdlib_logger.html#log_message-write-the-string-message-to-self-log_units)        | Subroutine | Sends a message
[`log_text_error`](./stdlib_logger.html#log_text_error-send-a-message-to-self-log_units-describing-an-error)     | Subroutine | Sends a message describing an error found in a line of text
[`log_units_assigned`](./stdlib_logger.html#log_units_assigned-returns-the-number-of-active-io-units) | Function   | Returns the number of active I/O units in `log_units`
[`log_warning`](./stdlib_logger.html#log_warning-write-the-string-message-to-log_units)        | Subroutine | Sends a message prepended by `'WARN: '`
[`remove_log_unit`](./stdlib_logger.html#remove_log_unit-remove-unit-from-self-log_units)    | Subroutine | Removes the `unit` number from the `log_units` array
```

Links to the specs of the different methods. Up to others to accept this suggestion or not.